Lib/idlelib/colorizer.py

```diff
@@ -16,6 +17,20 @@ def any(name, alternates):
 
 def make_pat():
     kw = r"\b" + any("KEYWORD", keyword.kwlist) + r"\b"
+    match_softkw = (
+        r"(^|(?<=\n))[ \t]*" +  # at beginning of string or after \n
+        r"(?P<PM_MATCH>match)\b" +
+        r"(?!([ \t]|\\\n)*[=,)\]}])"  # not followed by any of =,)]}
+    )
+    match_case_default = (
+        r"(^|(?<=\n))[ \t]*" +  # at beginning of string or after \n
+        r"(?P<PM_DEFAULT_CASE>case)[ \t]+(?P<PM_DEFAULT_UNDERSCORE>_)[ \t]*:"
```

Comment:
Looking for '\_' as a keyword is a nightmare. There are so many random cases (e.g. in a `sequence_pattern`) and there is no guarantee the start of the statement will be on the same line. Regexes, I believe, are way too limited for this. Maybe we should forget about '\_' for now?

Reply:
This is only specifically for `case _:`, and it seems to work well for that.

What other cases are nightmarish? (References would be helpful.)

Reply:
Yes, it does work well (as far as I can tell) with `case _:`. And I see your point about that being common so maybe we should keep it. One situation I was thinking of was:
```
case [[_],
      _]:
```
Here, both underscores should be highlighted as keywords.

Reply:
Perhaps we could mark any use of `_` in a `case` clause is such a soft-keyword?

Reply:
I believe that is a reasonable assumption. It seems you cannot use a variable named '\_' in a case statement. For example, the following raises a `SyntaxError` rather than `NameError`:

```
case [_ * 3]:
```

Reply:
The match doc in not completely clear to me but I believe it collectively says that isolated '_' is an unbindable keyword in case patterns.  Hence it cannot be used as a bound or binding name -- in patterns.  But this experiment show that it is normal elsewhere, even in match headers, guards, and actions.
```
>>> _ = 'a'
>>> match _:
...     case _ if _: print(_, 'ok')
... 
...     
a ok
```
If we can match any and every isolated '_' in patterns and only in patterns, great.  It does not matter if invalid (as after 'as').  I'm not sure if it is possible.

If we only wanted to match underscore in 'case  _:', we could do it more simply.
```suggestion
        r"(?P<CASE_UNDERSCORE>case[ \t]_)[ \t]*:"
```
Since by default and likely in most custom themes, the keyword background is the same as normal background, including whitespace in the tagged slice would not matter.

Reply:
`_ * 3` is invalid *because* `_` is a keyword with no value.  `_` is valid when used properly to mean 'match anything'.
```
>>> match ['a']:
...     case [_]: print('ok')
... 
ok
```
As I tried to say above, misuses that are misuses *because* it is parsed as a keyword should be tagged as a keyword.  In this case, tagging makes it more obvious that the error is because one is attempting to multiply a keyword rather than a value.